skbio/stats/composition.py

```diff
@@ -643,6 +643,113 @@ def ilr_inv(mat, basis=None, check=True):
 
 
 @experimental(as_of="0.4.0")
+def alr(mat, denominator_col = 0):
+    r"""
+    Performs additive log ratio transformation.
+    
+    
+    This function transforms compositions from a D-part Aitchison simplex to
+    a non-isometric real space of D-1 dimensions. The argument 
+    `denominator_col` defines the index of the column used as the common
+    denominator. The :math: `alr` transformed data are amenable to multivariate
+    analysis as long as statistics don't involve distances.
+
+    :math:`alr: S^D \rightarrow \mathbb{R}^{D-1}`
+
+    The alr transformation is defined as follows
+
+    .. math::
+        alr(x) = [log \frac{x_1}{x_D}, \ldots, log \frac{x_{D-1}}{x_D}]
+
+    where :math:`D` is the index of the part used as common denominator.
+
+    Parameters
+    ----------
+    mat: numpy.ndarray
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+
+    denominator_col: int
+       the index of the column of mat which should be used as the reference
+       composition
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skbio.stats.composition import alr
+    >>> x = np.array([.1, .3, .4, .2])
+    >>> alr(x)
+    array([ 1.09861229,  1.38629436,  0.69314718])
+    """
+    mat = closure(mat)
+    if len(mat.shape) == 2:
+        mat_t = mat.transpose()
+        num_col = list(range(0, mat_t.shape[0]))
+        del num_col[denominator_col]
+        lr = np.log(mat_t[num_col,:]/mat_t[denominator_col,:]).transpose()
+    elif len(mat.shape) == 1:
+        num_col = list(range(0, mat.shape[0]))
+        del num_col[denominator_col]
+        lr = np.log(mat[num_col]/mat[denominator_col])
+    else:
+        raise ValueError("mat must be either 1D or 2D")
+    return lr
+
+
+@experimental(as_of="0.4.0")
```

Comment:
should be `0.5.1-dev` since it isn't part of 0.5.1 (it's already released). part of the release process is to search and replace the dev versions